### ISSUE TITLE
Checkboxes in Tree Views need keyboard interaction design

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3573]
## Describe the bug

Currently, our trees do not support arrow key navigation. Once they do, actually including checkboxes in tree items will force us to create an overly complex keyboard interaction to operate them. Trees should gain focus as a whole and then arrow keys should be used to navigate through the tree items. If there are checkboxes inside of the tree item, we will have to design a keyboard interaction to break out of the tree navigation context and navigate within the tree item itself. There is some guidance about doing this for grid cells that we can possible adapt. See "Editing and Navigating Inside a Cell" at https://www.w3.org/TR/wai-aria-practices-1.1/#gridNav_inside.

Another solution, could be to simply show icons of checkboxes while using a standard Tree View implementation(https://www.w3.org/TR/wai-aria-practices-1.1/#TreeView). aria-selected can be used to logically convey the selected state while the icon of the checked checkbox conveys it visually. aria-describedby can be used to communicate the indeterminate state. But we don't have to do that either, since the indeterminate state is visual only(https://css-tricks.com/indeterminate-checkboxes/) but it could be beneficial to do so. 

## How to reproduce

https://stackblitz.com/github/clane/clarityTest2/tree/master/clarity-test-app

TBD

## Expected behavior

TBD

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

_Add any other notes about the problem here._
